tests/tls_e2e/common/utility.cpp

```diff
@@ -0,0 +1,318 @@
+#include <openenclave/internal/raise.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "tls_e2e_t.h"
+#include "utility.h"
+// clang-format on
+
+extern struct tls_control_args g_control_config;
+
+// input: input_data and input_data_len
+// output: key, key_size
+oe_result_t generate_key_pair(
+    uint8_t** public_key,
+    size_t* public_key_size,
+    uint8_t** private_key,
+    size_t* private_key_size)
+{
+    oe_result_t result = OE_FAILURE;
+    oe_asymmetric_key_params_t params;
+    char user_data[] = "test user data!";
+    size_t user_data_size = sizeof(user_data) - 1;
+
+    // Call oe_get_public_key_by_policy() to generate key pair derived from an
+    // enclave's seal key. If an enclave does not want to have this key pair
+    // tied to enclave instance, it can generate its own key pair using any
+    // chosen crypto API
+
+    params.type = OE_ASYMMETRIC_KEY_EC_SECP256P1; // MBEDTLS_ECP_DP_SECP256R1
+    params.format = OE_ASYMMETRIC_KEY_PEM;
+    params.user_data = user_data;
+    params.user_data_size = user_data_size;
+    result = oe_get_public_key_by_policy(
+        OE_SEAL_POLICY_UNIQUE,
+        &params,
+        public_key,
+        public_key_size,
+        NULL,
+        NULL);
+    OE_CHECK_MSG(
+        result,
+        "oe_get_public_key_by_policy(OE_SEAL_POLICY_UNIQUE) = %s",
+        oe_result_str(result));
+
+    result = oe_get_private_key_by_policy(
+        OE_SEAL_POLICY_UNIQUE,
+        &params,
+        private_key,
+        private_key_size,
+        NULL,
+        NULL);
+    OE_CHECK_MSG(
+        result,
+        "oe_get_private_key_by_policy(OE_SEAL_POLICY_UNIQUE) = %s",
+        oe_result_str(result));
+
+done:
+    return result;
+}
+
+// Compute the sha256 hash of given data.
+static int sha256(const uint8_t* data, size_t data_size, uint8_t sha256[32])
+{
+    int ret = 0;
+    mbedtls_sha256_context ctx;
+
+    mbedtls_sha256_init(&ctx);
+
+    ret = mbedtls_sha256_starts_ret(&ctx, 0);
+    if (ret)
+        goto done;
+
+    ret = mbedtls_sha256_update_ret(&ctx, data, data_size);
+    if (ret)
+        goto done;
+
+    ret = mbedtls_sha256_finish_ret(&ctx, sha256);
+    if (ret)
+        goto done;
+
+done:
+    mbedtls_sha256_free(&ctx);
+    return ret;
+}
+
+oe_result_t generate_certificate_and_pkey(
+    mbedtls_x509_crt* cert,
+    mbedtls_pk_context* private_key)
+{
+    oe_result_t result = OE_FAILURE;
+    uint8_t* output_cert = NULL;
+    size_t output_cert_size = 0;
+    uint8_t* private_key_buf = NULL;
+    size_t private_key_buf_size = 0;
+    uint8_t* public_key_buf = NULL;
+    size_t public_key_buf_size = 0;
+    int ret = 0;
+
+    result = generate_key_pair(
+        &public_key_buf,
+        &public_key_buf_size,
+        &private_key_buf,
+        &private_key_buf_size);
+    OE_CHECK_MSG(result, " failed with %s\n", oe_result_str(result));
+
+#if 0
+    OE_TRACE_INFO("private_key_buf_size:[%ld]\n", private_key_buf_size);
+#endif
+
```

Comment:
Remove the section lines:111 to 113

Reply:
This line will come in handy when debugging some signing issue, i will keep it.